### ISSUE TITLE
Support for the GraalVM

### ISSUE THREAD

**Issue by ShalokShalom** [matrix-org/synapse#14948]
GraalVM is a polyglot VM, that offers the JVM languages together with a variety of other languages, who are implemented within the so called Truffle framework:

https://www.graalvm.org/python/

The performance characteristics seem promising, and considering the overwhelming use of matrix.org as a homeserver and its use of Synapse, I think this is an interesting option.

The project supports 3.8.5 at the moment, and I could see a cooperation be fruitful for both 😊

**Comment by clokep** [comment]
Synapse uses quite a few C-extensions and has a built-in Rust extension using [PyO3](https://docs.rs/pyo3/); I have no idea how much this would impact using something like GraalVM. Some community members run Synapse on PyPy in the past. We won't be adding GraalVM as a supported target, although may be willing to merge some pull requests which make supporting it easier.

tl;dr Feel free to try it and let us know how it goes, but we do not have the resources to officially support running Synapse on GraalVM.

**Comment by ShalokShalom** [comment]
Works immediately: https://github.com/oracle/graal/tree/master/sulong

I dont know, how running Rust and C on Graal, performance wise, but considering the runtime characteristics of Python, I would consider this as a viable option. 

And then there is a FFI, I suppose.
